api/src/pcapi/core/offers/api.py

```diff
@@ -1273,3 +1273,53 @@ def delete_price_category(offer: models.Offer, price_category: models.PriceCateg
     validation.check_price_categories_deletable(offer)
     db.session.delete(price_category)
     db.session.commit()
+
+
+def reapproves_provider_product_and_rejected_offers(ean: str) -> bool:
+    product = models.Product.query.filter(
+        models.Product.extraData["ean"].astext == ean, models.Product.idAtProviders.is_not(None)
+    ).one_or_none()
+
+    if not product:
```

Comment:
Pour un ean donné il peut ne pas y avoir de produit mais exister des offres créées manuellement (celles qui n'ont pas pas un productId = product.id mais le extraData.ean ==ean).
Dans ce cas là les offres ne seront pas réactivées. 
Je pense que tu peux transformer le `if not product:` en `if not offers or not product` et le mettre après la récupération des offres

Reply:
Merci de cette remarque, effectivement j'ai oublié ce détail.
J'ai revu le code.
Néanmoins, sauf erreur de ma part, il faut que je complexifie un poil les conditions pour gérer le cas où il n'y a pas de produits MAIS que seulement des offres individuelles créées manuellement. Merci pour ta revue :smiley:

Reply:
Après discussions avec Patrick et Solène, nous avons besoin pour le moment que de gérer le cas de la réactivation d'un produit Titelive rejeté par les cgu ainsi que les offres liés. Si une offre est crée manuellement avec un ean qui serait non éligible vis à vis des cgu et donc à rejeter, n'est pas implémenter et necessite encore un cadrage. ça ne viendra pas maintenant. On doit pas réactiver les offres manuels si Titelive annonce que ce ean est valide.